biofefi/pages/3_Feature_Importance.py

```diff
@@ -237,64 +241,87 @@ def pipeline(
 experiment_name = experiment_selector(choices)
 base_dir = biofefi_experiments_base_dir()
 
-if experiment_name:
-    st.session_state[ConfigStateKeys.ExperimentName] = experiment_name
-
-    data_choices = os.listdir(base_dir / experiment_name)
-    data_choices = filter(lambda x: x.endswith(".csv"), data_choices)
 
-    model_choices = os.listdir(ml_model_dir(base_dir / experiment_name))
-    model_choices = [x for x in model_choices if x.endswith(".pkl")]
+if experiment_name:
 
-    explain_all_models = st.toggle(
-        "Explain all models", key=ConfigStateKeys.ExplainAllModels
+    fi_options = find_previous_fi_results(
```

Comment:
```suggestion
    previous_results_exist = find_previous_fi_results(
```
More meaningful name that reflects what you're doing.